data-structures.md

```diff
@@ -107,24 +109,45 @@ type Actor struct {
 	Head    Cid
     
     // Nonce is a counter of the number of messages this actor has sent
-	Nonce   Integer
+	Nonce   Uint64
     
     // Balance is this actors current balance of filecoin
-	Balance AttoFIL
+	Balance BigInteger
 }
 ```
 
+## State Tree
 
+The state trie keeps track of all state in Filecoin. It is effectively a map of addresses to `actors` in the system. It is implemented using a HAMT.
 
+## HAMT
 
-### Serialization
+TODO: link to spec for our CHAMP HAMT
 
-Actors are currently serialized simply by CBOR marshaling them, using lower-camel-cased field names.
 
-## State Tree
 
-The state trie keeps track of all state in Filecoin. It is effectively a map of addresses to `actors` in the system. It is implemented using a HAMT.
+# Filecoin Compact Serialization
 
-## HAMT
+Datastructures in Filecoin are encoded as compactly as is reasonable. At a high level, each object is converted into an ordered array of its fields (ordered by their appearance in the struct declaration), then CBOR marshaled, and prepended with an object type tag.
+
```

Comment:
this does allow for forward compatibility. Every object is tagged, if you change an object, you pick a new tag.

In general in Filecoin, old binaries will not be able to understand new protocol formats, and it won't make sense for them to try to do so. 

though maybe i can see a case where a thing wants to try and read messages, and if we were to add a new field for some weird reason, it wouldnt be able to read the old fields. I don't consider than much of an issue though, as my main concern is that the old impl *knows* the object is a different type, even if it can't understand it.